kernel/interrupts/src/aarch64/mod.rs

```diff
@@ -0,0 +1,416 @@
+use core::arch::global_asm;
+use core::cell::UnsafeCell;
+use core::ops::DerefMut;
+use core::fmt;
+
+use cortex_a::registers::*;
+
+use tock_registers::interfaces::Writeable;
+use tock_registers::interfaces::Readable;
+use tock_registers::registers::InMemoryRegister;
+
+use gic::{qemu_virt_addrs, ArmGic, IntNumber, Version as GicVersion};
+use irq_safety::{RwLockIrqSafe, MutexIrqSafe};
+use memory::get_kernel_mmi_ref;
+use log::{info, error};
+
+// Raw/Early exception handling is defined
+// in this assembly file.
```

Comment:
isn't this file just assembly-level exception handler trampolines / entry points? it's not actually early handling, right?